### PR TITLE
Add syntax highlighting and reformat models

### PR DESCRIPTION
**Pull request by Silmathoron**
This PR includes new files for syntax highlighting in geany and vscode.
Reformatted models to add spaces between numbers and units and switch ``/* */`` to ``""" """``.

I was also wondering if it would not be better to remove the ``/* */`` docstring since it is a relic from the dark ages ;)
Enforcing spaces between number and units might also be useful to facilitate syntax highlighting.

### CODE REVIEW

**Comment by Silmathoron** [comment]
@clinssen would it be possible to add the name of the file which fails in the tests? I new which one it was so it was easy, but otherwise the [error log](https://travis-ci.org/nest/nestml/jobs/563420314) is not most explicit ^^

File: pynestml/utils/logger.py [hunk 1]
```diff
@@ -100,8 +100,8 @@ def log_message(cls, neuron=None, code=None, message=None, error_position=None,
         from pynestml.meta_model.ast_source_location import ASTSourceLocation
         assert (neuron is None or isinstance(neuron, ASTNeuron)), \
             '(PyNestML.Logger) Wrong type of neuron provided (%s)!' % type(neuron)
-        assert (error_position is None or isinstance(error_position, ASTSourceLocation)), \
-            '(PyNestML.Logger) Wrong type of error position provided (%s)!' % type(error_position)
+        # assert (error_position is None or isinstance(error_position, ASTSourceLocation)), \
```
Comment (clinssen) [inline pynestml/utils/logger.py]: Why is this commented out?

File: extras/syntax-highlighting/geany/Readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Syntax highliting for Geany
```
Comment (clinssen) [inline extras/syntax-highlighting/geany/Readme.md]:
```suggestion
Syntax highlighting for Geany
```

File: extras/syntax-highlighting/geany/Readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+===========================
```
Comment (clinssen) [inline extras/syntax-highlighting/geany/Readme.md]:
```suggestion
=============================
```

File: extras/syntax-highlighting/visual-code/Readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Syntax highliting for Visual Studio Code
```
Comment (clinssen) [inline extras/syntax-highlighting/visual-code/Readme.md]:
```suggestion
Syntax highlighting for Visual Studio Code
```

File: extras/syntax-highlighting/visual-code/Readme.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+========================================
```
Comment (clinssen) [inline extras/syntax-highlighting/visual-code/Readme.md]:
```suggestion
==========================================
```

File: extras/syntax-highlighting/visual-code/nestml/language-configuration.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
+{
+	"comments": {
+		"lineComment": "#",
+		"blockComment": [ "\"\"\"", "\"\"\"" ]
```
Comment (clinssen) [inline extras/syntax-highlighting/visual-code/nestml/language-configuration.json]: Is there support for having more than one pair of open/close sequence, such that we can maintain support for the Java-style `/*...*/` comments?

File: extras/syntax-highlighting/visual-code/nestml/package.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
+    "name": "nestml",
+    "displayName": "NESTML",
+    "description": "NEST simulator markup language for model code-generation",
```
Comment (clinssen) [inline extras/syntax-highlighting/visual-code/nestml/package.json]:
```suggestion
    "description": "NEST simulator modeling language for spiking neural network code generation",
```

File: pynestml/utils/logger.py (hunk 1 above)
Comment (Silmathoron) [reply pynestml/utils/logger.py]: Ah, this is associated with #511, I forgot to remove it, I'll do that

File: extras/syntax-highlighting/visual-code/nestml/language-configuration.json (hunk 6 above)
Comment (Silmathoron) [reply extras/syntax-highlighting/visual-code/nestml/language-configuration.json]: I don't know... probably... but I hope we drop them anyway, so I would not want to investigate this too much until we reach an agreement ;)

File: extras/syntax-highlighting/visual-code/nestml/package.json (hunk 7 above)
Comment (Silmathoron) [reply extras/syntax-highlighting/visual-code/nestml/package.json]: Is this how we describe NESTML? We don't really generate code for "spiking neural network" but for neuron and synapse models, though, don't we?

Comment (clinssen) [reply extras/syntax-highlighting/visual-code/nestml/package.json]: Indeed, it's good to iterate a bit on this! This same phrase should re-appear in many places, e.g. the PyPI package, GitHub repository title, and so on.

"NESTML: A modeling language for spiking neurons and synapses"

Can we work the term "code generation" in there somewhere? Should we mention "point process"?

cc @jougs @abigailm

**Comment by jougs** [comment]
Any news on this?

File: pynestml/utils/logger.py (hunk 1 above)
Comment (clinssen) [reply pynestml/utils/logger.py]: Should it be removed our uncommented?

Comment (Silmathoron) [reply pynestml/utils/logger.py]: Unless you fixed it, I will probably have to remove it, but I'll check it again on both computers... not before the end of the week at best though...

File: extras/syntax-highlighting/visual-code/nestml/package.json (hunk 7 above)
Comment (clinssen) [reply extras/syntax-highlighting/visual-code/nestml/package.json]: @Silmathoron : we've come up with the following for now:

*NESTML: A concise modeling language and flexible code generator for spiking neuron and synapse models*

Would you be OK with this formulation?

Comment (Silmathoron) [reply extras/syntax-highlighting/visual-code/nestml/package.json]: Sure, I'll try to include it tomorrow

**Comment by Silmathoron** [comment]
@jougs @clinssen should be good to merge as soon as tests pass

File: models/iaf_cond_exp_sfa_rr.nestml [hunk 8]
```diff
@@ -113,12 +113,12 @@ neuron iaf_cond_exp_sfa_rr:
...
 end
 
-/*
+"""
 Name: iaf_cond_exp_sfa_rr_implicit - Simple conductance based leaky integrate-and-fire
                             neuron model.
 
 Description:
-iaf_cond_exp_sfa_rr is an iaf_cond_exp_sfa_rr i.e. an implementation of a
+iaf_cond_exp_sfa_rr is an iaf_cond_exp_sfa_rr i. e. an implementation of a
```
Comment (clinssen) [inline models/iaf_cond_exp_sfa_rr.nestml]:
```suggestion
iaf_cond_exp_sfa_rr is an iaf_cond_exp_sfa_rr i.e. an implementation of a
```

File: models/iaf_psc_alpha.nestml [hunk 9]
```diff
@@ -37,7 +37,7 @@ Remarks:
   components of the state vector and the non-zero matrix elements of
   the propagator.  Because the propagator is a lower triangular matrix
   no full matrix multiplication needs to be carried out and the
-  computation can be done "in place" i.e. no temporary state vector
+  computation can be done "in place" i. e. no temporary state vector
```
Comment (clinssen) [inline models/iaf_psc_alpha.nestml]:
```suggestion
  computation can be done "in place" i.e. no temporary state vector
```

File: models/iaf_psc_alpha.nestml [hunk 10]
```diff
@@ -51,7 +51,7 @@ Remarks:
   If tau_m is very close to tau_syn_ex or tau_syn_in, the model
   will numerically behave as if tau_m is equal to tau_syn_ex or
   tau_syn_in, respectively, to avoid numerical instabilities.
-  For details, please see IAF_Neruons_Singularity.ipynb in
+  For details, please see IAF_Neruons_Singularity. ipynb in
```
Comment (clinssen) [inline models/iaf_psc_alpha.nestml]:
```suggestion
  For details, please see IAF_Neruons_Singularity.ipynb in
```

File: models/iaf_psc_delta.nestml [hunk 11]
```diff
@@ -42,7 +42,7 @@ The present implementation uses individual variables for the
 components of the state vector and the non-zero matrix elements of
 the propagator.  Because the propagator is a lower triangular matrix
 no full matrix multiplication needs to be carried out and the
-computation can be done "in place" i.e. no temporary state vector
+computation can be done "in place" i. e. no temporary state vector
```
Comment (clinssen) [inline models/iaf_psc_delta.nestml]:
```suggestion
computation can be done "in place" i.e. no temporary state vector
```

File: models/iaf_psc_exp.nestml [hunk 12]
```diff
@@ -28,7 +28,7 @@ The present implementation uses individual variables for the
 components of the state vector and the non-zero matrix elements of
 the propagator.  Because the propagator is a lower triangular matrix
 no full matrix multiplication needs to be carried out and the
-computation can be done "in place" i.e. no temporary state vector
+computation can be done "in place" i. e. no temporary state vector
```
Comment (clinssen) [inline models/iaf_psc_exp.nestml]:
```suggestion
computation can be done "in place" i.e. no temporary state vector
```

File: models/iaf_psc_exp.nestml [hunk 13]
```diff
@@ -40,12 +40,12 @@ matrix objects.
 Remarks: If tau_m is very close to tau_syn_ex or tau_syn_in, the model
 will numerically behave as if tau_m is equal to tau_syn_ex or
 tau_syn_in, respectively, to avoid numerical instabilities.
-For details, please see IAF_Neruons_Singularity.ipynb in the
+For details, please see IAF_Neruons_Singularity. ipynb in the
```
Comment (clinssen) [inline models/iaf_psc_exp.nestml]:
```suggestion
For details, please see IAF_Neruons_Singularity.ipynb in the
```

File: models/iaf_psc_exp.nestml [hunk 14]
```diff
@@ -40,12 +40,12 @@ matrix objects.
...
 NEST source code (docs/model_details).
 
 iaf_psc_exp can handle current input in two ways: Current input
 through receptor_type 0 are handled as stepwise constant current
-input as in other iaf models, i.e., this current directly enters
+input as in other iaf models, i. e., this current directly enters
```
Comment (clinssen) [inline models/iaf_psc_exp.nestml]:
```suggestion
input as in other iaf models, i.e., this current directly enters
```

File: models/iaf_psc_exp.nestml [hunk 15]
```diff
@@ -93,16 +93,16 @@ neuron iaf_psc_exp:
...
-    I_e     pA = 0pA     # External current.
+    C_m pF = 250 pF       # Capacity of the membrane
+    tau_m ms = 10 ms      # Membrane time constant.
+    tau_syn_in ms = 2 ms  # Time constant of synaptic current.
+    tau_syn_ex ms = 2 ms  # Time constant of synaptic current.
+    t_ref ms = 2 ms       # Duration of refractory period
+    E_L  mV = -70 mV      # Resting potential.
+    function V_reset mV = -70 mV - E_L # reset value of the membrane potential
+    function Theta   mV = -55 mV - E_L # Threshold, RELATIVE TO RESTING POTENTAIL(!).
+                                   # I. e. the real threshold is (E_L_+V_th_)
```
Comment (clinssen) [inline models/iaf_psc_exp.nestml]:
```suggestion
                                   # I.e. the real threshold is (E_L_+V_th_)
```

File: models/iaf_tum_2000.nestml [hunk 16]
```diff
@@ -36,7 +36,7 @@ The present implementation uses individual variables for the
 components of the state vector and the non-zero matrix elements of
 the propagator.  Because the propagator is a lower triangular matrix
 no full matrix multiplication needs to be carried out and the
-computation can be done "in place" i.e. no temporary state vector
+computation can be done "in place" i. e. no temporary state vector
```
Comment (clinssen) [inline models/iaf_tum_2000.nestml]:
```suggestion
computation can be done "in place" i.e. no temporary state vector
```

File: models/iaf_tum_2000.nestml [hunk 17]
```diff
@@ -68,7 +68,7 @@ Remarks:
 If tau_m is very close to tau_syn_ex or tau_syn_in, the model
 will numerically behave as if tau_m is equal to tau_syn_ex or
 tau_syn_in, respectively, to avoid numerical instabilities.
-For details, please see IAF_Neruons_Singularity.ipynb in
+For details, please see IAF_Neruons_Singularity. ipynb in
```
Comment (clinssen) [inline models/iaf_tum_2000.nestml]:
```suggestion
For details, please see IAF_Neruons_Singularity.ipynb in
```

File: models/iaf_tum_2000.nestml [hunk 18]
```diff
@@ -118,9 +118,9 @@ neuron iaf_tum_2000:
     E_L     mV = -70 mV  # Resting potential.
     function V_reset mV = -70.0 mV - E_L # Reset value of the membrane potential
                                      # RELATIVE TO RESTING POTENTIAL(!).
-                                     # I.e. the real threshold is (V_reset + E_L).
+                                     # I. e. the real threshold is (V_reset + E_L).
```
Comment (clinssen) [inline models/iaf_tum_2000.nestml]:
```suggestion
                                     # I.e. the real threshold is (V_reset + E_L).
```

File: models/iaf_tum_2000.nestml [hunk 19]
```diff
@@ -118,9 +118,9 @@ neuron iaf_tum_2000:
...
     function V_th mV = -55.0 mV - E_L # Threshold, RELATIVE TO RESTING POTENTIAL(!).
-                                  # I.e. the real threshold is (E_L+V_th).
+                                  # I. e. the real threshold is (E_L+V_th).
```
Comment (clinssen) [inline models/iaf_tum_2000.nestml]:
```suggestion
                                  # I.e. the real threshold is (E_L+V_th).
```

File: models/mat2_psc_exp.nestml [hunk 20]
```diff
@@ -32,7 +32,7 @@ The present implementation uses individual variables for the
 components of the state vector and the non-zero matrix elements of
 the propagator. Because the propagator is a lower triangular matrix
 no full matrix multiplication needs to be carried out and the
-computation can be done "in place" i.e. no temporary state vector
+computation can be done "in place" i. e. no temporary state vector
```
Comment (clinssen) [inline models/mat2_psc_exp.nestml]:
```suggestion
computation can be done "in place" i.e. no temporary state vector
```

File: models/mat2_psc_exp.nestml [hunk 21]
```diff
@@ -68,9 +68,9 @@ neuron mat2_psc_exp:
   end
 
   initial_values:
-    V_abs mV  = 0mV # Membrane potential
+    V_abs mV  = 0 mV # Membrane potential
     function V_m mV = V_abs + E_L    # Relative membrane potential.
-                                      # I.e. the real threshold is (V_m-E_L).
+                                      # I. e. the real threshold is (V_m-E_L).
```
Comment (clinssen) [inline models/mat2_psc_exp.nestml]:
```suggestion
                                      # I.e. the real threshold is (V_m-E_L).
```

File: models/non_functional/iaf_psc_exp_multisynapse.nestml [hunk 22]
```diff
@@ -28,7 +28,7 @@ neuron iaf_psc_exp_multisynapse_neuron:
   initial_values:
     V_abs mV # membrane potential
     function V_m mV = V_abs + E_L # Relative membrane potential.
-                               # I.e. the real threshold is (V_m-E_L).
+                               # I. e. the real threshold is (V_m-E_L).
```
Comment (clinssen) [inline models/non_functional/iaf_psc_exp_multisynapse.nestml]:
```suggestion
                               # I.e. the real threshold is (V_m-E_L).
```

File: models/terub_gpe.nestml [hunk 23]
```diff
@@ -21,12 +21,12 @@ Problems/Todo:
   better spike detection
 
 References:
-  Terman, D. and Rubin, J.E. and Yew, A.C. and Wilson, C.J.
+  Terman, D. and Rubin, J. E. and Yew, A. C. and Wilson, C. J.
```
Comment (clinssen) [inline models/terub_gpe.nestml]:
```suggestion
  Terman, D. and Rubin, J.E. and Yew, A. C. and Wilson, C.J.
```

File: models/terub_gpe.nestml [hunk 24]
```diff
@@ -21,12 +21,12 @@ Problems/Todo:
...
   Activity Patterns in a Model for the Subthalamopallidal Network
   of the Basal Ganglia
   The Journal of Neuroscience, 22(7), 2963-2976 (2002)
 
-  Rubin, J.E. and Terman, D.
+  Rubin, J. E. and Terman, D.
```
Comment (clinssen) [inline models/terub_gpe.nestml]:
```suggestion
  Rubin, J.E. and Terman, D.
```

File: models/terub_gpe.nestml [hunk 25]
```diff
@@ -198,12 +198,12 @@ Problems/Todo:
   better spike detection
 
 References:
-  Terman, D. and Rubin, J.E. and Yew, A.C. and Wilson, C.J.
+  Terman, D. and Rubin, J. E. and Yew, A. C. and Wilson, C. J.
```
Comment (clinssen) [inline models/terub_gpe.nestml]:
```suggestion
  Terman, D. and Rubin, J.E. and Yew, A.C. and Wilson, C.J.
```

File: models/terub_gpe.nestml [hunk 26]
```diff
@@ -198,12 +198,12 @@ Problems/Todo:
...
   Activity Patterns in a Model for the Subthalamopallidal Network
   of the Basal Ganglia
   The Journal of Neuroscience, 22(7), 2963-2976 (2002)
 
-  Rubin, J.E. and Terman, D.
+  Rubin, J. E. and Terman, D.
```
Comment (clinssen) [inline models/terub_gpe.nestml]:
```suggestion
  Rubin, J.E. and Terman, D.
```

File: models/terub_stn.nestml [hunk 27]
```diff
@@ -21,12 +21,12 @@ Problems/Todo:
   better spike detection
 
 References:
-  Terman, D. and Rubin, J.E. and Yew, A.C. and Wilson, C.J.
+  Terman, D. and Rubin, J. E. and Yew, A. C. and Wilson, C. J.
```
Comment (clinssen) [inline models/terub_stn.nestml]:
```suggestion
  Terman, D. and Rubin, J.E. and Yew, A.C. and Wilson, C.J.
```

File: models/terub_stn.nestml [hunk 28]
```diff
@@ -21,12 +21,12 @@ Problems/Todo:
...
   Activity Patterns in a Model for the Subthalamopallidal Network
   of the Basal Ganglia
   The Journal of Neuroscience, 22(7), 2963-2976 (2002)
 
-  Rubin, J.E. and Terman, D.
+  Rubin, J. E. and Terman, D.
```
Comment (clinssen) [inline models/terub_stn.nestml]:
```suggestion
  Rubin, J.E. and Terman, D.
```

File: models/terub_stn.nestml [hunk 29]
```diff
@@ -207,12 +207,12 @@ Problems/Todo:
   better spike detection
 
 References:
-  Terman, D. and Rubin, J.E. and Yew, A.C. and Wilson, C.J.
+  Terman, D. and Rubin, J. E. and Yew, A. C. and Wilson, C. J.
```
Comment (clinssen) [inline models/terub_stn.nestml]:
```suggestion
  Terman, D. and Rubin, J.E. and Yew, A.C. and Wilson, C.J.
```

File: models/terub_stn.nestml [hunk 30]
```diff
@@ -207,12 +207,12 @@ Problems/Todo:
...
   Activity Patterns in a Model for the Subthalamopallidal Network
   of the Basal Ganglia
   The Journal of Neuroscience, 22(7), 2963-2976 (2002)
 
-  Rubin, J.E. and Terman, D.
+  Rubin, J. E. and Terman, D.
```
Comment (clinssen) [inline models/terub_stn.nestml]:
```suggestion
  Rubin, J.E. and Terman, D.
```

File: models/iaf_cond_exp_sfa_rr.nestml [hunk 31]
```diff
@@ -1,9 +1,9 @@
-/*
+"""
 Name: iaf_cond_exp_sfa_rr - Simple conductance based leaky integrate-and-fire
                             neuron model.
 
 Description:
-iaf_cond_exp_sfa_rr is an iaf_cond_exp_sfa_rr i.e. an implementation of a
+iaf_cond_exp_sfa_rr is an iaf_cond_exp_sfa_rr i. e. an implementation of a
```
Comment (clinssen) [inline models/iaf_cond_exp_sfa_rr.nestml]:
```suggestion
iaf_cond_exp_sfa_rr is an iaf_cond_exp_sfa_rr i.e. an implementation of a
```

File: models/hh_cond_exp_traub.nestml [hunk 32]
```diff
@@ -20,7 +20,7 @@ Only the channel variables m,h,n are implemented. The original
 contains variables called y,s,r,q and \chi.
 References:
 
-Traub, R.D. and Miles, R. (1991) Neuronal Networks of the Hippocampus.
+Traub, R. D. and Miles, R. (1991) Neuronal Networks of the Hippocampus.
```
Comment (clinssen) [inline models/hh_cond_exp_traub.nestml]:
```suggestion
Traub, R.D. and Miles, R. (1991) Neuronal Networks of the Hippocampus.
```

File: models/non_functional/iaf_psc_exp_multisynapse.nestml [hunk 33]
```diff
@@ -39,17 +39,17 @@ neuron iaf_psc_exp_multisynapse_neuron:
...
+    tau_syn ms[receptors] = 2 ms     # Time constant of synaptic current.
+    t_ref   ms = 2 ms    # Refractory period.
+    E_L     mV = 0 mV    # Resting potential.
+    function V_reset mV = -55.0 mV - E_L # Reset value of the membrane potential
                                      # RELATIVE TO RESTING POTENTIAL(!).
-                                     # I.e. the real threshold is (V_reset + E_L).
-    function V_th   mV = -70.0mV - E_L # Threshold, RELATIVE TO RESTING POTENTIAL(!).
-                                    # I.e. the real threshold is (E_L+V_th).
-    I_e     pA = 0pA     # External current.
+                                     # I. e. the real threshold is (V_reset + E_L).
```
Comment (Silmathoron) [inline models/non_functional/iaf_psc_exp_multisynapse.nestml]:
```suggestion
                                     # I.e. the real threshold is (V_reset + E_L).
```

File: models/non_functional/iaf_psc_exp_multisynapse.nestml [hunk 34]
```diff
@@ -39,17 +39,17 @@ neuron iaf_psc_exp_multisynapse_neuron:
...
+    function V_th   mV = -70.0 mV - E_L # Threshold, RELATIVE TO RESTING POTENTIAL(!).
+                                    # I. e. the real threshold is (E_L+V_th).
```
Comment (Silmathoron) [inline models/non_functional/iaf_psc_exp_multisynapse.nestml]:
```suggestion
                                    # I.e. the real threshold is (E_L+V_th).
```

File: models/iaf_psc_alpha.nestml (hunk 10 above)
Comment (Silmathoron) [inline models/iaf_psc_alpha.nestml]:
```suggestion
  For details, please see IAF_Neurons_Singularity.ipynb in
```

File: models/iaf_tum_2000.nestml (hunk 17 above)
Comment (Silmathoron) [inline models/iaf_tum_2000.nestml]:
```suggestion
For details, please see IAF_Neurons_Singularity.ipynb in
```

**Comment by Silmathoron** [comment]
Done ;)

**Pull request merged**